### PR TITLE
Add C++ runtime framework binaries to self-contained exe

### LINKED ISSUE #145: Add C++ runtime framework binaries to standalone exe
Winget-Create has a dependency on the [VC++ v14 Desktop Framework Package](https://docs.microsoft.com/troubleshoot/cpp/c-runtime-packages-desktop-bridge#how-to-install-and-update-desktop-framework-packages). Currently, the standalone exe does not include the binaries from this framework package, which means a user would have to ensure that the VC++ v14 Desktop Framework Package is already installed on the build machine prior to running Winget-Create. 

The standalone exe should be able to run without any dependencies so that it can be seamlessly integrated into any pipeline.

### PR DESCRIPTION
**Pull request by ryfu-msft**
Resolves #145 

Changes:
- Adds an additional ADO task to extract the binaries from the VCLibs appx package
- Modifies the csproj file to include binaries in build

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-create/pull/181)

### CODE REVIEW

File: pipelines/azure-pipelines.release.yml [hunk 1]
```diff
@@ -34,7 +34,6 @@ jobs:
           echo "##vso[task.setvariable variable=majorMinorBuildVersion;isOutput=true]$version"
         name: GetVersionStep
         displayName: Get version from CLI project
```
Comment (palenshus) [inline pipelines/azure-pipelines.release.yml:36]: Why delete this line?

File: pipelines/azure-pipelines.release.yml [hunk 2]
```diff
@@ -166,15 +181,13 @@ jobs:
                   "ToolVersion" : "1.0"
               }
             ]
```
Comment (palenshus) [inline pipelines/azure-pipelines.release.yml:183]: I'm seeing a bunch of newlines removed

File: pipelines/azure-pipelines.release.yml [hunk 3]
```diff
@@ -90,6 +88,23 @@ jobs:
...
           destination: '$(workingDirectory)\WingetCreateCLI\Telemetry\'
 
+      # Copies binary dependencies from VCLibs package to be included in standalone exe     
+      - task: PowerShell@2
+        displayName: Download and install VCLibs package
+        inputs:
+          targetType: 'inline'
+          script: | 
+            iwr https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx -OutFile Microsoft.VCLibs.x64.14.00.Desktop.appx
+            Add-AppxPackage Microsoft.VCLibs.x64.14.00.Desktop.appx
```
Comment (palenshus) [inline pipelines/azure-pipelines.release.yml:98]: Isn't this step unnecessary now?

File: src/WingetCreateCLI/WingetCreateCLI.csproj [hunk 4]
```diff
@@ -77,6 +77,19 @@
       <Generator>PublicResXFileCodeGenerator</Generator>
     </EmbeddedResource>
   </ItemGroup>
+    
+  <!--Include VCLib binary dependencies-->  
+  <ItemGroup Condition="'$(SelfContained)' == true">
```
Comment (palenshus) [inline src/WingetCreateCLI/WingetCreateCLI.csproj:82]: What's selfcontained mean?

Comment (palenshus) [inline src/WingetCreateCLI/WingetCreateCLI.csproj:82]: Will this fail when built outside the release pipeline?

**Review by palenshus — CHANGES_REQUESTED** [review]
:clock1:

File: pipelines/azure-pipelines.release.yml (hunk 1 above)
Comment (ryfu-msft) [reply pipelines/azure-pipelines.release.yml]: reverted file to remove line changes

File: pipelines/azure-pipelines.release.yml (hunk 2 above)
Comment (ryfu-msft) [reply pipelines/azure-pipelines.release.yml]: reverted file to remove line changes

File: pipelines/azure-pipelines.release.yml (hunk 3 above)
Comment (ryfu-msft) [reply pipelines/azure-pipelines.release.yml]: yes, I missed this. Removed

File: src/WingetCreateCLI/WingetCreateCLI.csproj (hunk 4 above)
Comment (ryfu-msft) [reply src/WingetCreateCLI/WingetCreateCLI.csproj:82]: selfcontained is the flag that indicates whether to publish the .NET runtimes with the exe. I figured we would want the standalone exe to be able to run without any dependencies.

Comment (ryfu-msft) [reply src/WingetCreateCLI/WingetCreateCLI.csproj:82]: I didn't have any issues building this outside of the release. Either in vs or using dotnet publish in terminal with or without those files present.

Comment (palenshus) [reply src/WingetCreateCLI/WingetCreateCLI.csproj:82]: Ooh gotcha, that's why the build succeeds when run locally, makes sense

**Review by palenshus — APPROVED** [review]
:shipit:

**Pull request merged**